### PR TITLE
Sync origin packages with scheduler

### PR DESCRIPTION
**Pull request by chefsalim**
This change updates the builder services to synchronize changes to the origin packages from the builder-originsrv to the builder-scheduler service. When the packages are synchronized, the dependency graph is also extended appropriately.

Signed-off-by: Salim Alam <salam@chef.io>

giphy 23

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-originsrv/src/data_store.rs [hunk 1]
```diff
@@ -864,6 +869,40 @@ fn sync_origins(pool: Pool) -> DbResult<EventOutcome> {
     Ok(result)
 }
 
+fn sync_packages(pool: Pool) -> DbResult<EventOutcome> {
+    println!("Sync packages called");
```
Comment (reset) [inline components/builder-originsrv/src/data_store.rs]: Might want to swap this to debug output

File: components/builder-jobsrv/src/data_store.rs [hunk 2]
```diff
@@ -298,7 +298,7 @@ impl DataStore {
     /// * If the dispatched jobs cannot be selected from the database
     pub fn reset_jobs(&self) -> Result<()> {
         let conn = self.pool.get_shard(0)?;
-        let rows = &conn.query("SELECT reset_jobs_v1()", &[])
+        &conn.query("SELECT reset_jobs_v1()", &[])
```
Comment (reset) [inline components/builder-jobsrv/src/data_store.rs]: I don't think you need the `&` here

**Comment by reset** [comment]
tenor-162570578

@chefsalim looks great! A few super minor comments inline, feel free to make changes and merge or ignore them and just merge on your own ;)

**Comment by chefsalim** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/226610199).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/226610199).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**